content/scalers/redis-streams.md

```diff
@@ -0,0 +1,147 @@
+
+One of its features includes [`Consumer Groups`](https://redis.io/topics/streams-intro#consumer-groups), that allows a group of clients to co-operate consuming a different portion of the same stream of messages.
+
+This specification describes the `redis-streams` trigger that scales based on the *Pending Entries List* (see [`XPENDING`](https://redis.io/commands/xpending)) for a specific Consumer Group of a Redis Stream.
+
+
+```yaml
+triggers:
+- type: redis-streams
+  metadata:
+    address: REDIS_SERVER # Required if host and port are not provided. Format - host:port
+    host: REDIS_HOST # Required if address is not provided
+    port: REDIS_PORT # Required if address is not provided and host has been provided
+    password: REDIS_PASSWORD # optional (can also use authenticationRef)
+    stream: my-stream # Required - name of the Redis Stream
+    consumerGroup: my-consumer-group # Required - name of consumer group associated with Redis Stream
+    pendingEntriesCount: "10" # Required - number of entries in the Pending Entries List for the specified consumer group in the Redis Stream
+    enableTLS: "false" # optional
+    databaseIndex: "0" # optional
+```
+
+**Parameter list:**
+
+- `address`: Name of the environment variable your deployment uses to get the Redis server URL. The resolved host should follow a format like `my-redis:6379`.
+   - This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
+
+> As an alternative to the `address` field, the user can specify `host` and `port` parameters. 
+
+- `host`: Name of the environment variable your deployment uses to get the Redis server host. 
+    - This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
+
+> It is not required if `address` has been provided
+
+- `port`: Name of the environment variable your deployment uses to get the Redis server port. 
+   - This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
+
+> It is only to be used along with the `host` attribute and not required if `address` has been provided
+
+- `password` (optional): Name of the environment variable your deployment uses to get the Redis password.
+   - This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
+
+- `stream`: Name of the Redis Stream
+
+- `consumerGroup`: Name of the Consumer group associated with Redis Stream
+
+- `pendingEntriesCount`: Threshold for the number of `Pending Entries List`. This is the average target value to scale the workload. Defaults to `5` 
+
+- `databaseIndex`: The Redis database index. Defaults to `0` if not specified
+
+- `enableTLS`: Set this to `true` if TLS connection to Redis is required. Defaults to `false`
+
+### Authentication Parameters
+
+The scaler supports two modes of authentication:
+
+**`password` metadata field**
```

Comment:
Suggestion - Use header so that it shows up in side navigation
```suggestion
#### Using password authentication
```

Reply:
updated as per feedback